python_modules/dagster/dagster/core/asset_defs/asset_partitions.py

```diff
@@ -0,0 +1,39 @@
+from dagster.core.definitions.events import AssetKey
+
+from .asset import AssetsDefinition
+from .partition_key_range import PartitionKeyRange
+
+
+def get_parent_partitions(
```

Comment:
maybe `get_parent_partitions_for_partition_range`?

Also, we have to specify asset key here because the assets definition might contain multiple asset keys? 

Do we anticipate this function being used outside of test files?  Something doesn't quite feel right about having this just be a utility function.

Reply:
`get_parent_partitions_for_partition_range` sounds good to me.

I do anticipate this function being used outside of test files - it will be necessary for the backend calls that back operations like "refresh this partition and all upstream partitions" or "determine whether any of the upstream partitions of this partition have changed more recently than it has".

I suspect that it might ultimately make sense as a method on some sort of `AssetGraph` or `AssetRepository` class or something, but IMO it's fine as a free-standing function for now.